src/main/java/com/idrsolutions/microservice/FormVuServlet.java

```diff
@@ -114,7 +116,29 @@ protected void convert(String uuid,
 
         //Makes the directory for the output file
         new File(outputDirStr + "/" + fileNameWithoutExt).mkdirs();
+        final int pageCount;
+        try {
+            final PdfDecoderServer decoder = new PdfDecoderServer(false);
+            decoder.openPdfFile(inputFile.getAbsolutePath());
 
+            decoder.setEncryptionPassword(conversionParams.getOrDefault("org.jpedal.pdf2html.password", ""));
+
+            if (decoder.isEncrypted() && !decoder.isPasswordSupplied()) {
+                LOG.log(Level.SEVERE, "Invalid Password");
+                DBHandler.getInstance().setError(uuid, 1070, "Invalid password supplied.");
+                return;
+            }
+
```

Comment:
@KieranFrance Do you have any thoughts on potentially moving the isForm check to be handled here instead of creating a new instance of decoder again? (lines 146-155)
```suggestion

            if (!decoder.isForm()) {
                LOG.log(Level.SEVERE, "PDF file does not contain any forms.");
                DBHandler.getInstance().setError(uuid, 1220, "Provided PDF file does not contain any forms.");
                return;
            }

```